rsa/pkcs1.py

```diff
@@ -283,6 +279,28 @@ def sign(message, priv_key, hash):
     return block
 
 
+def sign(message, priv_key, hash_method):
+    """Signs the message with the private key.
+
+    Hashes the message, then signs the hash with the given key. This is known
+    as a "detached signature", because the message itself isn't altered.
+
+    :param message: the message to sign. Can be an 8-bit string or a file-like
+        object. If ``message`` has a ``read()`` method, it is assumed to be a
+        file-like object.
+    :param priv_key: the :py:class:`rsa.PrivateKey` to sign with
+    :param hash_method: the hash method used on the message. Use 'MD5', 'SHA-1',
+        'SHA-256', 'SHA-384' or 'SHA-512'.
+    :return: a message signature block.
+    :raise OverflowError: if the private key is too small to contain the
+        requested hash.
+
+    """
+
+    # Calculate the hash and perform the signing
+    return sign_hash(hash(message, hash_method), priv_key, hash_method)
```

Comment:
Please splitt his up into two lines, using a helper variable. The way it is now it does too much on one line; a traceback referring to an error in this line will be too ambiguous.